### PR TITLE
PIPE2D-804: Simplify the broken-shutter analytic-corrections code

### PR DESCRIPTION
**Pull request by RobertLuptonTheGood**

### CODE REVIEW

File: python/lsst/obs/pfs/isrTask.py [hunk 1]
```diff
@@ -68,6 +73,66 @@ class PfsIsrTaskConfig(ipIsr.IsrTaskConfig):
...
+                for k, v in cls.__cache.items():
+                    if k_min is None or v[1] < n_min:
+                        k_min = k
+                        n_min = v[1]
+
+                if k_min is not None:
+                    print(f"Clearing cached kernel for {k_min}")
+                    del cls.__cache[k_min]
+
+            print(f"Computing ikern for {key} ({t_exp}s)")
```
Comment (hassanxp) [inline python/lsst/obs/pfs/isrTask.py:106]: Can the ``print`` call be converted to a ``log.info`` call? Similar occurrences later in the class (except the ``show()`` method which is fine as-is)

File: python/lsst/obs/pfs/isrTask.py [hunk 2]
```diff
@@ -68,6 +73,66 @@ class PfsIsrTaskConfig(ipIsr.IsrTaskConfig):
...
+
+class BrokenShutterKernelCache:
+    __cache = {}
+    __maxsize = 0
+
+    def __init__(self, maxsize=0):
+        self.setCacheSize(maxsize)
+
+    @classmethod
+    def compute(cls, kern, t_exp):
```
Comment (hassanxp) [inline python/lsst/obs/pfs/isrTask.py:91]: Missing docString.

File: python/lsst/obs/pfs/isrTask.py [hunk 3]
```diff
@@ -237,3 +266,53 @@ def runDataRef(self, sensorRef):
             isrQa.writeThumbnail(sensorRef, result.flattenedThumb, "flattenedThumb")
 
         return result
+
+    def run(self, exposure, *args, **kwargs):
```
Comment (hassanxp) [inline python/lsst/obs/pfs/isrTask.py:270]: Missing docString

**Review by hassanxp — APPROVED** [review]
Changes look fine. Only minor comments on docString and print statements. These can be deferred to a later ticket if these changes need to be put into master promptly.

File: python/lsst/obs/pfs/isrTask.py (hunk 1 above)
Comment (RobertLuptonTheGood) [reply python/lsst/obs/pfs/isrTask.py:106]: Well, not trivially as the cache has no associated log or parent, so it'd be stand-alone --- that's why I didn't do it with logging in the first place.  But sure, I can see how hard that would be to implement.

File: python/lsst/obs/pfs/isrTask.py (hunk 2 above)
Comment (RobertLuptonTheGood) [reply python/lsst/obs/pfs/isrTask.py:91]: I think it's barely worth adding, but I'll add something that does a little more than echoing the method names.

**Pull request merged**